[2024-11-22T19:03:03.125+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-11-22T19:03:03.262+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T19:01:00+00:00 [queued]>
[2024-11-22T19:03:03.301+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T19:01:00+00:00 [queued]>
[2024-11-22T19:03:03.302+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 2
[2024-11-22T19:03:03.413+0000] {taskinstance.py:2888} INFO - Executing <Task(BranchPythonOperator): process_transaction> on 2024-11-22 19:01:00+00:00
[2024-11-22T19:03:03.473+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=996) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-11-22T19:03:03.477+0000] {standard_task_runner.py:72} INFO - Started process 1000 to run task
[2024-11-22T19:03:03.478+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'fraud_detection', 'process_transaction', 'scheduled__2024-11-22T19:01:00+00:00', '--job-id', '472', '--raw', '--subdir', 'DAGS_FOLDER/fraud_detection_dag.py', '--cfg-path', '/tmp/tmp9eocwogd']
[2024-11-22T19:03:03.486+0000] {standard_task_runner.py:105} INFO - Job 472: Subtask process_transaction
[2024-11-22T19:03:03.618+0000] {task_command.py:467} INFO - Running <TaskInstance: fraud_detection.process_transaction scheduled__2024-11-22T19:01:00+00:00 [running]> on host c6af8dbaf1e5
[2024-11-22T19:03:03.935+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='fraud_team' AIRFLOW_CTX_DAG_ID='fraud_detection' AIRFLOW_CTX_TASK_ID='process_transaction' AIRFLOW_CTX_EXECUTION_DATE='2024-11-22T19:01:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-11-22T19:01:00+00:00'
[2024-11-22T19:03:03.937+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-11-22T19:03:03.962+0000] {baseoperator.py:405} WARNING - BranchPythonOperator.execute cannot be called outside TaskInstance!
[2024-11-22T19:03:03.987+0000] {fraud_detection_dag.py:77} INFO - Données récupérées depuis XCom: {"columns":["cc_num","merchant","category","amt","first","last","gender","street","city","state","zip","lat","long","city_pop","job","dob","trans_num","merch_lat","merch_long","is_fraud","current_time"],"index":[545686],"data":[[3586008444788268,"fraud_Medhurst PLC","shopping_net",8.41,"Crystal","Fuller","F","000 Jennifer Mills","Issaquah","WA",98027,47.4974,-122.0107,50489,"Osteopath","1984-02-29","6327000daad6d36e8c87efc7e11af78f",47.877028,-122.632127,0,1732302179236]]}
[2024-11-22T19:03:03.990+0000] {fraud_detection_dag.py:85} INFO - Données JSON parsées avec succès
[2024-11-22T19:03:03.994+0000] {fraud_detection_dag.py:89} INFO - DataFrame créé avec 1 lignes
[2024-11-22T19:03:03.996+0000] {fraud_detection_dag.py:93} INFO - Colonnes renommées avec succès
[2024-11-22T19:03:03.996+0000] {fraud_detection_dag.py:94} INFO - Colonnes disponibles: ['cc_num', 'merchant', 'category', 'amt', 'first', 'last', 'gender', 'street', 'city', 'state', 'zip', 'lat', 'long', 'city_pop', 'job', 'dob', 'trans_num', 'merch_lat', 'merch_long', 'is_fraud', 'trans_date_trans_time']
[2024-11-22T19:03:03.999+0000] {fraud_detection_dag.py:98} INFO - Conversion de trans_date_trans_time en unix_time réussie
[2024-11-22T19:03:04.038+0000] {fraud_detection_dag.py:107} INFO - Transformation des données...
[2024-11-22T19:03:05.428+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-22T19:03:05.456+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/sklearn/ensemble/_base.py:172: UserWarning: Loky-backed parallel loops cannot be called in a multiprocessing, setting n_jobs=1
  n_jobs = min(effective_n_jobs(n_jobs), n_estimators)

[2024-11-22T19:03:05.482+0000] {fraud_detection_dag.py:119} INFO - Prediction de fraude: NORMAL, Probabilité: 41.61%
[2024-11-22T19:03:05.560+0000] {fraud_detection_dag.py:125} INFO - Prédiction: NORMAL (Probabilité: 41.61%)
[2024-11-22T19:03:05.561+0000] {python.py:240} INFO - Done. Returned value was: notify_normal
[2024-11-22T19:03:05.562+0000] {branch.py:38} INFO - Branch into notify_normal
[2024-11-22T19:03:05.563+0000] {skipmixin.py:230} INFO - Following branch ('notify_normal',)
[2024-11-22T19:03:05.605+0000] {skipmixin.py:278} INFO - Skipping tasks [('skip_processing', -1), ('notify_fraud', -1)]
[2024-11-22T19:03:05.724+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-11-22T19:03:05.734+0000] {taskinstance.py:352} INFO - Marking task as SUCCESS. dag_id=fraud_detection, task_id=process_transaction, run_id=scheduled__2024-11-22T19:01:00+00:00, execution_date=20241122T190100, start_date=20241122T190303, end_date=20241122T190305
[2024-11-22T19:03:05.939+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 0
[2024-11-22T19:03:06.032+0000] {taskinstance.py:3900} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2024-11-22T19:03:06.036+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
